clock/scheduledJobs/jobCleanup.js

```diff
@@ -53,46 +69,45 @@ function execute(batchSize, delay) {
 
     // get n jobs
     return kue.Job.rangeByStateAsync('complete', from, to, 'asc')
-    .catch((err) => {
-      console.log('Error getting completed jobs from queue', err);
-      return Promise.reject(err);
-    })
+      .catch((err) => {
+        console.log('Error getting completed jobs from queue', err);
+        return Promise.reject(err);
+      })
+
+      // delete n jobs
+      .then((jobs) =>
+        Promise.all(jobs.map((job) => {
+          if (delay > 0 && now - job.updated_at < delay) {
+            return 'skipped';
+          }
 
-    // delete n jobs
-    .then((jobs) =>
-      Promise.all(jobs.map((job) => {
-        if (delay > 0 && now - job.updated_at < delay) {
-          return 'skipped';
-        } else {
           return job.removeAsync()
-          .catch((err) => {
-            console.log(`Error removing job ${job.id}`, err);
-            return Promise.reject(err);
-          });
-        }
-      }))
-    )
+            .catch((err) => {
+              errorJobCount++;
+              return Promise.reject(err);
```

Comment:
Where is this rejection being caught and handled?
(Hint: if we're not handling it somewhere, it probably shouldn't be a rejection!)

Reply:
Same here... just added the ++... Should I change to resolve?

Reply:
nah, don't change anything now since you didn't just write this now

we already have a user story rid of all our UnhandledPromiseRejectionWarning from tests and in prod so we'll just deal with it as part of that